File: frogment/src/main/java/ch/grze/frogment/frogment/FragmentCallbacks.java
package ch.grze.frogment.frogment;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import ch.grze.frogment.core.Core;
import ch.grze.frogment.core.callbacks.AbstractFragmentLifecycleCallbacks;

public class FragmentCallbacks extends AbstractFragmentLifecycleCallbacks {
    public FragmentCallbacks(Core core) {
        super(core);
    }

    @Override
    public void onFragmentPreCreated(FragmentManager fm, Fragment f, Bundle savedInstanceState) {
        super.onFragmentPreCreated(fm, f, savedInstanceState);

        try {
            final StateAwareFrogmentInterface stateAwareFrogment = getTypedFragment(f);

            stateAwareFrogment.getStateAwareFrogmentComponent().onFragmentPreCreated(savedInstanceState);
        } catch (ClassCastException ignored) {}
    }

    @Override
    public void onFragmentActivityCreated(FragmentManager fm, Fragment f, Bundle savedInstanceState) {
        super.onFragmentActivityCreated(fm, f, savedInstanceState);

        try {
            final StateAwareFrogmentInterface stateAwareFrogment = getTypedFragment(f);

            stateAwareFrogment.getStateAwareFrogmentComponent().onFragmentActivityCreated(savedInstanceState);
        } catch (ClassCastException ignored) {}
    }

    @Override
    public void onFragmentSaveInstanceState(FragmentManager fm, Fragment f, Bundle outState) {
        super.onFragmentSaveInstanceState(fm, f, outState);

        try {
            final StateAwareFrogmentInterface stateAwareFrogment = getTypedFragment(f);

            stateAwareFrogment.getStateAwareFrogmentComponent().onFragmentSaveInstanceState(outState);
        } catch (ClassCastException ignored) {}
    }
}

File: frogment/src/main/java/ch/grze/frogment/frogment/FragmentCallbacks.kt
package ch.grze.frogment.frogment

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import ch.grze.frogment.core.Core
import ch.grze.frogment.core.callbacks.AbstractFragmentLifecycleCallbacks

class FragmentCallbacks(core: Core) : AbstractFragmentLifecycleCallbacks(core) {
    override fun onFragmentPreCreated(fm: FragmentManager?, f: Fragment?, savedInstanceState: Bundle?) {
        super.onFragmentPreCreated(fm, f, savedInstanceState)

        (f as? StateAwareFrogmentInterface<*>)
            ?.stateAwareFrogmentComponent
            ?.onFragmentPreCreated(savedInstanceState)
    }

    override fun onFragmentActivityCreated(fm: FragmentManager?, f: Fragment?, savedInstanceState: Bundle?) {
        super.onFragmentActivityCreated(fm, f, savedInstanceState)

        (f as? StateAwareFrogmentInterface<*>)
            ?.stateAwareFrogmentComponent
            ?.onFragmentActivityCreated(savedInstanceState)
    }

    override fun onFragmentSaveInstanceState(fm: FragmentManager?, f: Fragment?, outState: Bundle) {
        super.onFragmentSaveInstanceState(fm, f, outState)

        (f as? StateAwareFrogmentInterface<*>)
            ?.stateAwareFrogmentComponent
            ?.onFragmentSaveInstanceState(outState)
    }
}